### PR TITLE
Updates StsCredentialsProvider to explicitly handle AwsSessionCredentials

### PR DESCRIPTION
**Pull request by cenedhryn**
## Motivation and Context
The abstract base class for STS credentials provider was [recently made public](#4045). This change changes the protected `getUpdatedCredentials` method to return `AwsSessionCredentials`, a public class, instead of instead of the STS model class `Credentials`, so that the SDK handles and stores credentials in a type which may be modified by the SDK. The change is not strictly backwards compatible, but given the short timespan between the two PRs it's unlikely that more than the occasional user would be affected. 

## Modifications
- Changes the protected `getUpdatedCredentials` method to return `AwsSessionCredentials` instead of the STS model class `Credentials`.
- Updates all known subclasses of `StsCredentialsProvider`. 
- Adds builder and `expirationTime` to `AwsSessionCredentials` in order to convert STS credentials to session credentials. Previously, expiration has been stored outside of the session credentials
- Removes `SessionCredentialsHolder` which has been rendered obsolete by this change

## Testing
Unit tests added.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [ ] I have read the [CONTRIBUTING](https://github.com/aws/aws-sdk-java-v2/blob/master/CONTRIBUTING.md) document
- [ ] Local run of `mvn install` succeeds
- [ ] My code follows the code style of this project
- [ ] My change requires a change to the Javadoc documentation
- [ ] I have updated the Javadoc documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed
- [ ] I have added a changelog entry. Adding a new entry must be accomplished by running the `scripts/new-change` script and following the instructions. Commit the new file created by the script in `.changes/next-release` with your changes.
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License

- [ ] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

File: .changes/next-release/feature-AWSSDKforJavav2AWSSTS-d87c45d.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+    "type": "feature",
+    "category": "AWS SDK for Java v2, AWS STS",
+    "contributor": "",
+    "description": "Updates the core STS credential provider logic to return AwsSessionCredentials instead of STS Credentials, and adds expirationTime to AwsSessionCredentials\""
```
Comment (millems) [inline .changes/next-release/feature-AWSSDKforJavav2AWSSTS-d87c45d.json:5]: nit: extra quote

File: services/sts/src/main/java/software/amazon/awssdk/services/sts/internal/StsAuthUtils.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.auth.credentials.AwsSessionCredentials;
+import software.amazon.awssdk.services.sts.model.Credentials;
+
+@SdkInternalApi
+public final class StsAuthUtils {
+    private StsAuthUtils() {
+    }
+
+    public static AwsSessionCredentials fromStsCredentials(Credentials credentials) {
```
Comment (gosar) [inline services/sts/src/main/java/software/amazon/awssdk/services/sts/internal/StsAuthUtils.java:27]: nit: I think `toAwsSessionCredentials` might be a better name. The "from" type is evident by the method parameter. The "to" type is more useful in the method name.

File: .changes/next-release/feature-AWSSDKforJavav2AWSSTS-d87c45d.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
+    "type": "feature",
+    "category": "AWS SDK for Java v2, AWS STS",
```
[the remark this answers is not in the document]
Comment (cenedhryn) [reply .changes/next-release/feature-AWSSDKforJavav2AWSSTS-d87c45d.json:3]: Makes sense, will update

File: services/sts/src/main/java/software/amazon/awssdk/services/sts/internal/StsAuthUtils.java (hunk 2 above)
Comment (cenedhryn) [reply services/sts/src/main/java/software/amazon/awssdk/services/sts/internal/StsAuthUtils.java]: agree, it read better

**Review by gosar — APPROVED** [review]
LGTM

**Review by millems — APPROVED** [review]

**Pull request merged**